src/modules/75-cd/components/EnvironmentGroups/EnvironmentGroupsFilters/EnvironmentGroupsFilters.tsx

```diff
@@ -0,0 +1,251 @@
+  )
+
+  const handleFilterSelection = (
+    option: SelectOption,
+    event?: React.SyntheticEvent<HTMLElement, Event> | undefined
+  ) => {
+    event?.stopPropagation()
+    event?.preventDefault()
+
+    if (option.value) {
+      updateQueryParams({
+        filterIdentifier: option.value.toString(),
+        filters: [] as any /* this will remove the param */
+      })
+    } else {
+      updateQueryParams({
+        filterIdentifier: [] as any /* this will remove the param */,
+        filters: [] as any /* this will remove the param */
+      })
+    }
+  }
+
+  const onApply = (inputFormData: FormikProps<EnvironmentGroupFilterFormType>['values']) => {
+    if (!isObjectEmpty(inputFormData)) {
+      const filterFromFormData = {
+        ...(inputFormData.envGroupName && {
+          envGroupName: inputFormData.envGroupName
+        }),
+        ...(inputFormData.description && {
+          description: inputFormData.description
+        }),
+        ...(inputFormData.envGroupTags && {
+          envGroupTags: Object.keys(inputFormData.envGroupTags || {})?.map((key: string) => {
+            return { key, value: inputFormData?.envGroupTags?.[key] } as NGTag
+          })
+        }),
+        ...(inputFormData.environments?.length && {
+          envIdentifiers: inputFormData.environments?.map((env: MultiSelectOption) => env?.value)
+        })
+      }
+      updateQueryParams({ page: [] as any, filters: JSON.stringify({ ...(filterFromFormData || {}) }) })
+      hideFilterDrawer()
+    } else {
+      showError(getString('filters.invalidCriteria'))
+    }
+  }
+
+  const handleSaveOrUpdate = async (
+    isUpdate: boolean,
+    data: FilterDataInterface<EnvironmentGroupFilterFormType, FilterInterface>
+  ) => {
+    setLoading(true)
+    const requestBodyPayload = createRequestBodyPayload({
+      isUpdate,
+      data,
+      projectIdentifier,
+      orgIdentifier
+    })
+
+    const saveOrUpdateHandler = filterRef.current?.saveOrUpdateFilterHandler
+    if (saveOrUpdateHandler && typeof saveOrUpdateHandler === 'function') {
+      const updatedFilter = await saveOrUpdateHandler(isUpdate, requestBodyPayload)
+      updateQueryParams({ filters: JSON.stringify({ ...(updatedFilter || {}) }) })
+    }
+
+    setLoading(false)
+    refetchFilters()
+  }
+
+  const handleDelete = async (filterIdentifier: string) => {
+    setLoading(true)
+    const deleteHandler = filterRef.current?.deleteFilterHandler
+    if (deleteHandler && typeof deleteFilter === 'function') {
+      await deleteHandler(filterIdentifier)
+    }
+    setLoading(false)
+
+    if (filterIdentifier === appliedFilter?.identifier) {
+      reset()
+    }
+    refetchFilters()
+  }
+
+  const handleFilterClick = (filterIdentifier: string) => {
+    if (filterIdentifier !== UNSAVED_FILTER_IDENTIFIER) {
+      updateQueryParams({
+        filterIdentifier,
+        filters: [] as any
```

Comment:
Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9264786)
```suggestion
        filters: [] as unknown
```